modules/friendlytag.js

```diff
@@ -147,18 +160,59 @@ Twinkle.tag.callback = function friendlytagCallback() {
 	Window.display();
 
 	if (Twinkle.tag.mode === "article") {
+
+		Twinkle.tag.alreadyPresentTags = [];
+
+		if (Twinkle.tag.untaggable) {
+			// Look for existing maintenance tags in the lead section and put them in array
+
+			// All tags are HTML table elements that are direct children of .mw-parser-output,
+			// except when they are within {{multiple issues}}
+			$('.mw-parser-output').children().each( function parsehtml(i,e) {
+
+				// break out on encountering the first heading, which means we are no
+				// longer in the lead section
+				if(e.tagName === 'H2')
+					return false;
+
+				// All tags have their first class name beginning with "box-"
+				if(e.className.indexOf('box-') === 0) {
```

Comment:
Sort of related to my above comment, but exactly how unique is `box-` as a class?  It's present in the [AfD template(https://en.wikipedia.org/wiki/Template_talk:Article_for_deletion/dated#Any_use_for_name_parameter%3F) but I worry about others; CSD and PROD are clear.

Reply:
It's not at all unique. Every template derived from Module:Message box, that has a `|name=` parameter set, will have a class beginning with `box-`. This is somewhat intentional, as this means _any_ tag that has a valid `|name=` can be removed using Twinkle. 

If there's consensus, I can add an additional `if (Twinkle.tag.article.tags[tag])` condition before pushing `tag` into `Twinkle.tag.alreadyPresentTags`, which would mean only the 80-odd tags identified by TW are removable. I don't see the point of doing this, though.

Reply:
Okay, I had been thinking we'd limit it to the Twinkle set.  I suppose you're right again, but I'm anxious about what that means for the list; I can deal with AfD but there may be some others we don't want there.  This would make the list of potential removable tags anything in templatespace that transcludes Ambox and uses `name`, right?  I can dwell on this a bit and go through that list.

Reply:
That's right...

Reply:
See https://github.com/azatoth/twinkle/pull/485#issuecomment-486888615